src/engine/controls/enginecontrol.h

```diff
@@ -57,40 +57,48 @@ class EngineControl : public QObject {
 
     virtual void setEngineMaster(EngineMaster* pEngineMaster);
     void setEngineBuffer(EngineBuffer* pEngineBuffer);
-    virtual void setCurrentSample(const double dCurrentSample,
-            const double dTotalSamples, const double dTrackSampleRate);
+    virtual void setFrameInfo(mixxx::audio::FramePos currentPosition,
+            mixxx::audio::FramePos trackEndPosition,
+            mixxx::audio::SampleRate sampleRate);
     QString getGroup() const;
 
-    void setBeatLoop(double startPosition, bool enabled);
-    void setLoop(double startPosition, double endPosition, bool enabled);
+    void setBeatLoop(mixxx::audio::FramePos startPosition, bool enabled);
+    void setLoop(mixxx::audio::FramePos startPosition,
+            mixxx::audio::FramePos endPosition,
+            bool enabled);
 
     // Called to collect player features for effects processing.
     virtual void collectFeatureState(GroupFeatureState* pGroupFeatures) const {
         Q_UNUSED(pGroupFeatures);
     }
 
     /// Called whenever a seek occurs to allow the EngineControl to respond.
-    virtual void notifySeek(double dNewPlaypos) {
-        Q_UNUSED(dNewPlaypos);
+    virtual void notifySeek(mixxx::audio::FramePos position) {
+        Q_UNUSED(position);
     };
 
     virtual void trackLoaded(TrackPointer pNewTrack);
     virtual void trackBeatsUpdated(mixxx::BeatsPointer pBeats);
 
   protected:
-    struct SampleOfTrack {
-        double current;
-        double total;
-        double rate;
+    struct FrameInfo {
```

Comment:
`SampleOfTrack` was a POD without any implicitly generated C++ ctors/dtor. Did you check if this has any implications when using it as `ControlValueAtomic<FrameInfo>`?

We should at least explicitly add constructors for direct initialization of all members.

Reply:
This is struct, so it lacks the `m_` prefix for members. Should I add them? Or should i use a prefix for parameters instead? Otherwise the constructors arguments would have the same name as the members.

Reply:
I guess you don't need any disambiguation for this case since C++11.

Reply:
`ControlRingValue` should work with any copyable type.